app/assets/javascripts/discourse/app/controllers/topic.js

```diff
@@ -1700,25 +1700,37 @@ export default Controller.extend(bufferedProperty("model"), {
             postNumberDifference > 0 &&
             postNumberDifference < 7
           ) {
-            this._scrollToPost(data.post_number);
+            this._scrollToDiscobotPost(data.post_number);
           }
         }
       },
       this.get("model.message_bus_last_id")
     );
   },
 
-  _scrollToPost(postNumber) {
+  _scrollToDiscobotPost(postNumber) {
     discourseDebounce(
       this,
       function () {
-        const $post = $(`.topic-post article#post_${postNumber}`);
+        const post = document.querySelector(
+          `.topic-post article#post_${postNumber}`
+        );
 
-        if ($post.length === 0 || isElementInViewport($post[0])) {
+        if (!post || isElementInViewport(post)) {
           return;
         }
 
-        $("html, body").animate({ scrollTop: $post.offset().top }, 1000);
+        const headerOffset =
+          parseInt(
+            getComputedStyle(document.body).getPropertyValue("--header-offset"),
+            10
+          ) || 0;
+        const viewportOffset = post.getBoundingClientRect();
```

Comment:
Note that this can leverage `domUtils` from https://github.com/discourse/discourse/pull/15313 once that PR is merged.